File: Baekjoon/Silver/Silver3_2579.py
import sys
input = sys.stdin.readline

n = int(input())
arr = [[0, 0, 0]]
value = [0] * (n+1)
for i in range(1, n+1):
    arr.append(int(input())) # (계단의 점수, 올라간 총 점수, 연속된 계단 수)

if len(arr)==2:
    print(arr[1])

elif len(arr)==3:
    print(arr[1] + arr[2])

else:
    value[1] = arr[1]
    value[2] = arr[2] + arr[1]

    for i in range(3, n+1):
        value[i] = max(value[i-2] + arr[i], value[i-3] + arr[i-1] + arr[i])

    print(value[n])

File: Baekjoon/Silver/Silver3_2579.java
package codingTest;

import java.io.*;
import java.util.*;

public class common {

    public static void main(String[] args) throws IOException {
        int n;
        int ans;
        int[] arr;
        int[] dp;

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());

        arr = new int[n + 1];
        dp = new int[n + 1];

        for (int i = 1; i <= n; i++) {
            st = new StringTokenizer(br.readLine());
            arr[i] = Integer.parseInt(st.nextToken());
        }

        if (n == 1) {
            System.out.println(arr[1]);
            return;
        }

        dp[1] = arr[1];
        dp[2] = arr[1] + arr[2];

        for (int i = 3; i <= n; i++) {
            dp[i] = Math.max(dp[i - 2] + arr[i], dp[i - 3] + arr[i - 1] + arr[i]);
        }


        System.out.println(dp[n]);
    }
}